pkg/epp/datalayer/endpoint_pool.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package datalayer
+
+import (
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/common"
+)
+
+type EndpointPool struct {
+	EndpointMeta           *EndpointsMeta
+	DisableK8sCrdReconcile bool
+	GKNN                   common.GKNN
```

Comment:
Do we really need to track this here? This type captures the name and kind of a k8s resource, but it is used to encode the pool type. Do we actually need to know the source of the pool?